src/engine-actions/translate-swap.js

```diff
@@ -44,6 +44,84 @@ function calculateTimeLock (extendedTimeLockDelta, secondsPerBlock, blockHeight)
   return maxTimeLockDeltaInBlocks.plus(blockHeight).toString()
 }
 
+/**
+ * gRPC error code for a CANCELLED request, either by a client or a server
+ * @see {@link https://github.com/grpc/grpc/blob/master/doc/statuscodes.md}
+ * @constant
+ * @type {Number}
+ */
+const CANCELLED_CODE = 1
+
+/**
+ * Use the sendToRoute function as though it returned a promise
+ * @param  {String}      paymentHash     Base64 string of the payment hash to pay for
+ * @param  {Array}       routes          Routes to use
+ * @param  {Object}      options.logger
+ * @param  {grpc.Client} options.client
+ * @return {Promise<TranslateSwapOutcome>}
+ */
+function sendToRouteSync (paymentHash, routes, { logger, client }) {
+  return new Promise((resolve, reject) => {
+    try {
+      logger.debug('Setting up stream to sendToRoute', { paymentHash })
+      const stream = sendToRoute({ client })
+
+      // keep local state of whether we cancelled so that we can
```

Comment:
[nit] whether we manually cancelled the stream?

Reply:
Eh, I think the "manual" isn't really the distinction, it's just whether it was cancelled by the client or the server.